proposals/lip-0022.md

```diff
@@ -177,25 +177,19 @@ Internally, the algorithm works as follows:
 
 In the case that `ListStandbyDelegates` contains less than 2 elements, the delegates with rank 102 and 103 will be chosen as `delegate1 `and `delegate2` respectively.
 
-### Round Computation and Delegate List Ordering
+### Round Computation and Delegate List
 
 The round computation is performed in the same way as in the current implementation with the difference that `delegate1` and `delegate2`, as defined in the previous section, are included in the forging delegate list of `round` + 1. This implies the following:
 
-*   The round length is now 103 block slots. Currently, this is defined by the constant  <code>[ACTIVE_DELEGATES](https://github.com/LiskHQ/lisk-sdk/blob/310283408102ff9f891830225afe8124e833a608/config/default/constants.js#L55)</code>.  For clarity, this constant should be replaced for three constants defining the number of the top 101 delegates by `delegateWeight`, the number of standby delegates, i.e., 2, and the length of the round (sum of the two previous constants).
+*   The round length is now 103 block slots. Currently, this is defined by the constant  <code>[ACTIVE_DELEGATES](https://github.com/LiskHQ/lisk-sdk/blob/310283408102ff9f891830225afe8124e833a608/config/default/constants.js#L55)</code>.  For clarity, this constant can be replaced for three constants defining the number of the top 101 delegates by `delegateWeight`, the number of standby delegates, i.e., 2, and the length of the round (sum of the two previous constants).
 *   The change in the length of the round affects the logic to check when a round is finished which is 103 block slots in total.
-*   `delegate1` and `delegate2` have to be included in the delegate list<sup>3</sup> before it is re-ordered for `round` + 1. Currently, the function <code>[getKeysSortByVote](https://github.com/LiskHQ/lisk-sdk/blob/7c31c9d4b12a72f18a35ead79198b2ba6d4bd091/framework/src/modules/chain/submodules/delegates.js#L32)</code> computes this list before being re-ordered.
+*   `delegate1` and `delegate2` have to be included in the delegate list<sup>3</sup> before it is re-ordered for `round` + 1. Currently, the function <code>[createRoundDelegateList](https://github.com/LiskHQ/lisk-sdk/blob/5997714fde5181701fa0508eebd7f3c595cbf92c/elements/lisk-dpos/src/delegates_list.ts#L148)</code> computes this list before being re-ordered.
 
-Bear in mind that this proposal does not change any other point in the round processing once the list with the 103 forging delegates for the next round is computed.  
+This LIP also proposes to use the 20 bytes adddress of a delegate as defined in [LIP-0018](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0018.md) for the [delegate lists](https://github.com/LiskHQ/lisk-sdk/blob/5997714fde5181701fa0508eebd7f3c595cbf92c/elements/lisk-dpos/src/delegates_list.ts#L124). This also implies two things:
```

Comment:
This line and the two next line rather refer to the implementation than the protocol if I understand correctly. I don't understand why here the order in the delegate list is important because the forging order is defined in LIP 0003.

Reply:
With the first bulletpoint I wanted to cover every case in which there is a tie by `delegateWeight`. Bear in mind that this is not about the ordered output of the function in LIP 0003 but about the input of said function. As said in line 186 here, a list of 103  delegates have to be created were 101 of them are the top 101 delegates by `delegateWeight`. The bulletpoint here covers this situation saying that if for example delegate 101 and delegate 102 have the same `delegateWeight`, the one with the "smaller" address will make it to the next forging round. This is now done by the function `createRoundDelegateList` but it is done with the delegate public keys. Perhaps it is clearer if I join both points in the same bullet point at line 186?

Reply:
Regarding the second bulletpoint at line 190, I think it is in the limit of being considered a protocol change. I preferred to state it here to avoid future issues.

Reply:
Good point regarding the tie-breaking. Maybe the "Round Computation and Delegate List" section could be re-structures a bit to make this more clear. One way would be to define the delegate list from scratch (or forger list as used in the documentation) and how it is created (add 101 active delegate by weight with tie-breaking by address plus 2 randomly selected standby delegates). Currently, it rather defines how an object (delegate_list class) in the code needs to be modified.